py-polars/src/expr/general.rs

```diff
@@ -1049,6 +1049,22 @@ impl PyExpr {
             .with_fmt("extend")
             .into()
     }
+    fn prepend_constant(&self, py: Python, value: Wrap<AnyValue>, n: usize) -> Self {
+        let value = value.into_py(py);
+        self.inner
+            .clone()
+            .apply(
```

Comment:
We don't want to use apply anymore for expressions. They have to use the current expression architecture with `apply_private`.

Reply:
Thanks @ritchie46 . I'm still pretty new to rust and a  still learning the codebase so this may take me a little while to figure out.

Reply:
I would say try tranlating the `apply` into `apply_private` and see where the compiler brings you. Then you should hit the enum that implements all expressions.